### ISSUE TITLE
Linked de Bruijn graph for HaplotypeCaller and Mutect2

### ISSUE THREAD

**Issue by droazen** [broadinstitute/gatk#5828]
Based on the paper by @kvg here: https://academic.oup.com/bioinformatics/article/34/15/2556/4938484

**Comment by droazen** [comment]
Notes from @kvg :

```
Manuscript on LdBG details: https://academic.oup.com/bioinformatics/article/34/15/2556/4938484

Reference implementation of the assembler in C: https://github.com/mcveanlab/mccortex
mcveanlab/mccortex

My Java library for accessing and manipulating LdBGs and associated formats: https://github.com/mcveanlab/CortexJDK

A useful starting point is the tests I've written to produce Figure 1 (the one with the pentagram cycle) from the manuscript.

Without links: https://github.com/mcveanlab/CortexJDK/blob/4ba64ee268314729c871916dfc9ee7c9c422c5cb/public/java/tests/uk/ac/ox/well/cortexjdk/utils/traversal/TraversalEngineTest.java#L210

With links: https://github.com/mcveanlab/CortexJDK/blob/4ba64ee268314729c871916dfc9ee7c9c422c5cb/public/java/tests/uk/ac/ox/well/cortexjdk/utils/traversal/TraversalEngineTest.java#L229

 (Oh and FYI, there's one place where my implementation of the read threading currently doesn't match the McCortex C reference implementation - the handling of sequencing errors and replacing the errorful kmers with kmers from the graph.  It's an easy thing to add; I just hadn't gotten around to it because I didn't have the need to do that alignment myself given that all my graphs and links were constructed with McCortex anyway.)
```

**Comment by davidbenjamin** [comment]
Closed by #6394
